### PR TITLE
Cap 0027: first class multiplexed accounts

### PR DESCRIPTION
**Pull request by MonsieurNicolas**
# Description

This PR implements [CAP-0027 : First-class multiplexed accounts](https://github.com/stellar/stellar-protocol/blob/master/core/cap-0027.md)

This PR addresses feedback from #2497 

# Checklist
- [ ] Reviewed the [contributing](https://github.com/stellar/stellar-core/blob/master/CONTRIBUTING.md#submitting-changes) document
- [ ] Rebased on top of master (no merge commits)
- [ ] Ran `clang-format` v5.0.0 (via `make format` or the Visual Studio extension)
- [ ] Compiles
- [ ] Ran all tests
- [ ] If change impacts performance, include supporting evidence per the [performance document](https://github.com/stellar/stellar-core/blob/master/performance-eval/performance-eval.md)

### CODE REVIEW

File: src/transactions/PathPaymentStrictReceiveOpFrame.cpp [hunk 1]
```diff
@@ -47,7 +47,7 @@ PathPaymentStrictReceiveOpFrame::doApply(AbstractLedgerTxn& ltx)
         return false;
     }
     innerResult().success().last = SimplePaymentResult(
-        getDestID(), getDestAsset(), mPathPayment.destAmount);
+        getDestMuxedAccount(), getDestAsset(), mPathPayment.destAmount);
```
Comment (jonjove) [inline src/transactions/PathPaymentStrictReceiveOpFrame.cpp:50]: This change isn't mentioned in the proposal. Is this change actually desired?

File: src/transactions/PathPaymentStrictSendOpFrame.cpp [hunk 2]
```diff
@@ -104,8 +104,8 @@ PathPaymentStrictSendOpFrame::doApply(AbstractLedgerTxn& ltx)
     {
         return false;
     }
-    innerResult().success().last =
-        SimplePaymentResult(getDestID(), getDestAsset(), maxAmountSend);
+    innerResult().success().last = SimplePaymentResult(
+        getDestMuxedAccount(), getDestAsset(), maxAmountSend);
```
Comment (jonjove) [inline src/transactions/PathPaymentStrictSendOpFrame.cpp:108]: Same comment as for `PathPaymentStrictReceipeOpFrame`.

File: src/xdr/Stellar-transaction.x [hunk 3]
```diff
@@ -967,7 +1008,7 @@ struct InnerTransactionResult
     case txBAD_AUTH_EXTRA:
     case txINTERNAL_ERROR:
     case txNOT_SUPPORTED:
-    // txFEE_BUMP_INNER_FAILED is not included
+        // txFEE_BUMP_INNER_FAILED is not included
```
[the remark this answers is not in the document]
Comment (MonsieurNicolas) [reply src/xdr/Stellar-transaction.x:1011]: yes this is clang-format. It looks like it was not run for a while on the .x files

File: src/transactions/PathPaymentStrictReceiveOpFrame.cpp (hunk 1 above)
Comment (MonsieurNicolas) [reply src/transactions/PathPaymentStrictReceiveOpFrame.cpp:50]: yeah I saw this too. I don't think we can actually do this either as SDKs are probably using the return the RAW result (like exchanges checking if a transaction is for them).
This is not a conversation for the PR but against the CAP, so let's discuss elsewhere.

Comment (MonsieurNicolas) [reply src/transactions/PathPaymentStrictReceiveOpFrame.cpp:50]: Oh it was actually mentioned. I missed it as well (last bullet point).

```
The following fields, which were previously an AccountID or AccountID*, are now a MuxedAccount or MuxedAccount* (respectively):

    PaymentOp::destination
    PathPaymentStrictReceiveOp::destination
    PathPaymentStrictSendOp::destination
    Operation::sourceAccount
    Operation::destination (for ACCOUNT_MERGE)
    Transaction::sourceAccount
    SimplePaymentResult::destination
```

**Comment by jonjove** [comment]
r+ ef71e8c

**Pull request merged**